UnityProject/Assets/Scripts/SpriteHandler/SpriteHandler.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+
+public class SpriteHandler : NetworkBehaviour
+{
+	public SpriteRenderer spriteRenderer;
+
+	public List<SpriteList> spriteList = new List<SpriteList>();
+	[SyncVar(hook = nameof(SyncSprite))] public int spriteIndex;
+
+	private List<SpriteInfo> infoList = new List<SpriteInfo>();
+	private int animationIndex = 0;
+	private float timeElapsed = 0;
+	private float waitTime;
+
+	public override void OnStartClient()
+	{
+		SyncSprite(spriteIndex);
+	}
+
```

Comment:
it's hard to reproduce, but if this object is pooled, its animation when spawned will be initially set to whatever it had when it was put back into the pool. You should define a hook for OnSpawnedServer which resets the sprite to its correct initial value (probably need a new initialSpriteIndex field for assigning this in editor, as mentioned earlier).